cmd/modern/root/install/mssql-base.go

```diff
@@ -468,17 +480,55 @@ CHECK_POLICY=OFF`
 	}
 }
 
+func getEscapedDbName(dbName string) string {
+	dbName = strings.ReplaceAll(dbName, "'", "''")
+	dbName = strings.ReplaceAll(dbName, "]", "]]")
+	return dbName
+}
+
+//parseDbName returns the databaseName from --using arg
+// It sets database name to the specified database name
+// or in absence of it, it is set to the filename without
+// extension.
+func parseDbName(usingDbUrl string) string {
+	u, _ := url.Parse(usingDbUrl)
+	dbToken := path.Base(u.Path)
+	if dbToken != "." && dbToken != "/" {
+		lastIdx := strings.LastIndex(dbToken, ".bak")
+		if lastIdx != -1 {
+			//Get file name without extension
+			fileName := dbToken[0:lastIdx]
+			lastIdx += 5
+			if lastIdx >= len(dbToken) {
+				return fileName
+			}
+			//Return database name if it was specified
+			return dbToken[lastIdx:]
+		}
+	}
+	return ""
+}
+
+func extractUrl(usingArg string) string {
+	urlEndIdx := strings.LastIndex(usingArg, ".bak")
+	if urlEndIdx != -1 {
+		return usingArg[0:(urlEndIdx + 4)]
+	}
+	return ""
+}
+
 func (c *MssqlBase) downloadAndRestoreDb(
 	controller *container.Controller,
 	containerId string,
 	userName string,
 ) {
 	output := c.Cmd.Output()
+	databaseName := parseDbName(c.usingDatabaseUrl)
+	databaseUrl := extractUrl(c.usingDatabaseUrl)
 
-	u, err := url.Parse(c.usingDatabaseUrl)
+	u, err := url.Parse(databaseUrl)
```

Comment:
>Parse(databaseUrl) [](http://example.com/codeflow?start=15&length=18)

parsedbName is already calling url.Parse so why do we need to do it again? 
Can parseDbName return all the relevant Url attributes ?

Reply:
This was only needed for printing the u.Hostname()
I think we need not return URL attributes just for this case since we are already validating the URL as part of validateUsingUrlExists() which gets called before even we download the file.
I will remove this code which requires us to parse the URL again.